pkg/nfd-master/nfd-master.go

```diff
@@ -883,13 +914,24 @@ func (m *nfdMaster) refreshNodeFeatures(cli k8sclient.Interface, node *corev1.No
 	maps.Copy(extendedResources, crExtendedResources)
 	extendedResources = m.filterExtendedResources(features, extendedResources)
 
+	if m.config.Restrictions.DisableExtendedResources {
+		klog.InfoS("maximum number taints per CR restriction is applied, skipping")
+		extendedResources = map[string]string{}
+	}
```

Comment:
Maybe move this to `filterExtendedResources()`, similarly to labels and annotations?

Also, adjust the logging

```suggestion
	if len(extendedResources) > 0 && m.config.Restrictions.DisableExtendedResources {
		klog.V(2).InfoS("extended resources are disabled in configuration (restrictions.disableExtendedResources=true)")
		extendedResources = map[string]string{}
	}
```